keyboards/ergodox_ez/keymaps/nfriend/keymap.c

```diff
@@ -0,0 +1,1095 @@
+/* Copyright (C) 2020 Nathan Friend
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "version.h"
+#include "keymap_german.h"
+#include "keymap_nordic.h"
+#include "keymap_french.h"
+#include "keymap_spanish.h"
+#include "keymap_hungarian.h"
+
+#define KC_MAC_UNDO LGUI(KC_Z)
+#define KC_MAC_CUT LGUI(KC_X)
+#define KC_MAC_COPY LGUI(KC_C)
+#define KC_MAC_PASTE LGUI(KC_V)
+#define KC_PC_UNDO LCTL(KC_Z)
+#define KC_PC_CUT LCTL(KC_X)
+#define KC_PC_COPY LCTL(KC_C)
+#define KC_PC_PASTE LCTL(KC_V)
+#define ___ KC_TRNS
+
+/* Shortcut for Option+Ctrl */
+#define OC(keycode) LALT(LCTL(keycode))
+
+/* Base layers */
+#define COLEMAK 0
+#define QWERTY 1
+#define DVORAK 2
+
+/* Keys the base layers share */
+#define COMMON 3
+
+/* OS layers */
+#define MAC 4
+#define WINDOWS 5
+
+/* Symbols */
+#define NUMPAD 6
+#define SYMBOLS 7
+
+/* Window management */
+#define WM_MAC 8
+#define WM_WINDOWS 9
+
+/* Navigation */
+#define ARROWS_MAC 10
+#define ARROWS_WINDOWS 11
+
+/* Misc */
+#define MISC 13
```

Comment:
We generally recommend using enums for layer names. As well as prepending them with underscores. 

```suggestion
enum layer_names {
    COLEMAK,
    QWERTY,
    DVORAK,
    COMMON,
    MAC,
    WINDOWS,
    NUMPAD,
    SYMBOLS,
    WM_MAC,
    WM_WINDOWS,
    ARROWS_MAC,
    ARROWS_WINDOWS,
    MISC,
};
```